### ISSUE TITLE
Does not work with Typescript

### ISSUE THREAD

**Issue by danawoodman** [UnwrittenFun/prettier-plugin-svelte#33]
It appears this fails when using `<script lang="ts">`.

```shell
npm i -D prettier prettier-plugin-svelte
npx prettier --write ./**/*.svelte
```

Generates the following errors. I used [this Typescript template](https://github.com/pyoner/svelte-ts-template) with no modifications.

```
src/App.svelte
[error] src/App.svelte: TypeError: require(...).parse is not a function
[error]     at Object.parse (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier-plugin-svelte/plugin.js:532:47)
[error]     at Object.parse$2 [as parse] (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:10635:19)
[error]     at coreFormat (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:13844:23)
[error]     at format (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:14102:73)
[error]     at formatWithCursor (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:14118:12)
[error]     at Object.formatWithCursor (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:42529:15)
[error]     at format$1 (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:43898:21)
[error]     at /Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:44093:16
[error]     at /Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:44033:14
[error]     at Array.forEach (<anonymous>)
src/Counter.svelte
[error] src/Counter.svelte: TypeError: require(...).parse is not a function
[error]     at Object.parse (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier-plugin-svelte/plugin.js:532:47)
[error]     at Object.parse$2 [as parse] (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:10635:19)
[error]     at coreFormat (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:13844:23)
[error]     at format (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:14102:73)
[error]     at formatWithCursor (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:14118:12)
[error]     at Object.formatWithCursor (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:42529:15)
[error]     at format$1 (/Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:43898:21)
[error]     at /Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:44093:16
[error]     at /Users/danawoodman/code/playground/svelte-app/node_modules/prettier/bin-prettier.js:44033:14
[error]     at Array.forEach (<anonymous>)
```

**Comment by taylorzane** [comment]
The issue is actually with the version of svelte specified in the `svelte-ts-template`. (The `parse` function wasn't available until [v3.2.0](https://github.com/sveltejs/svelte/blob/master/CHANGELOG.md#320))

`prettier-plugin-svelte` specifies that version as a minimum peer dependency. Though that warning does kind of get lost in the noise of an install...

```sh
taylor 🐳 npm i -D prettier prettier-plugin-svelte
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN prettier-plugin-svelte@0.5.1 requires a peer of svelte@^3.2.0 but none is installed. You must install peer dependencies yourself.
# ... other logs ...
```

You should be able to upgrade `svelte` manually and get a successful format. It might also be a good idea to bug the maintainer to update the reference to at least be a carat dependency rather than a hard dependency on version `3.0.0`.

Hope this helps!

**Comment by UnwrittenFun** [comment]
Closing this as I believe what @taylorzane said is correct. Feel free to reopen if still having issues
